storage/src/device.rs

```diff
@@ -402,10 +402,18 @@ pub trait BlobChunkInfo: Any + Sync + Send {
 }
 
 /// An enumeration to encapsulate different [BlobChunkInfo] implementations for [BlobIoDesc].
+/// This helps to feed unified IO description to storage subsystem from both rafs v6 and v5 since
+/// rafs v6 have a different ChunkInfo definition on bootstrap.
 #[derive(Clone)]
 pub enum BlobIoChunk {
+    // For rafs v6 to pass chunk info to storage module.
+    // (blob_index, chunk_index)
     Address(u32, u32),
+    // TODO: `Address` is is always converted to `Base` for v6, it is a little confusing.
```

Comment:
For rafs v6, the chunk info may be stored in the data blob instead of bootstrap blob. So only the blob reader from the storage subsystem could convert `Address` into `BlobChunkInfo`.

Reply:
Yes. I am thinking about removing `Base(dyn BlobChunkinfo)` out from `BlobIoChunk`. We can define `BlobIoChunk` as a medium type defining how this IO requesting backend storage data varies on v5 and v6(without decompression control info). Then trait `BlobChunkinfo` is the core structure in **storage** sub-system related to a single backend chunk IO. Then we don't have nested type conversion any more. **storage** subsystem should be capable to generate `dyn BlobChunkinfo` from `BlobIoChunk`